engine/collection/compliance/core.go

```diff
@@ -275,7 +283,7 @@ func (c *Core) processBlockAndDescendants(proposal *cluster.Block) error {
 		return nil
 	}
 	for _, child := range children {
-		cpr := c.processBlockAndDescendants(child.Message)
+		cpr := c.processBlockAndDescendants(originID, child.Message)
```

Comment:
### ⚠️ (continued)
I think this is the problem because we probably received `child` from a node other than `origin`. In fact, `children` already is a slice 
 `[]flow.Slashable[*cluster.Block]` but we just disregard the value of `child.OriginID`.

I think the possibility for such bugs could be greatly reduced if we consistently used `proposal flow.Slashable[*cluster.Block]`. I feel the logic is already largely compatible, and we would only need to consolidate  the APIs. E.g. [method `processBlockAndDescendants(..)`](https://github.com/onflow/flow-go/blob/339fcc04b1575c86f4c092780167c009d44469a7/engine/collection/compliance/core.go#L239) would have the following signature
```golang
func (c *Core) processBlockAndDescendants(proposal flow.Slashable[*cluster.Block]) error
```

then we could write here
```suggestion
		cpr := c.processBlockAndDescendants(child)
```
Please see my next comment for a more details on my proposal for API cleanup.